src/main/java/de/zalando/aruha/nakadi/partitioning/PartitionsCache.java

```diff
@@ -0,0 +1,38 @@
+package de.zalando.aruha.nakadi.partitioning;
+
+import de.zalando.aruha.nakadi.exceptions.ServiceUnavailableException;
+import de.zalando.aruha.nakadi.repository.kafka.KafkaFactory;
+import kafka.common.KafkaException;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class PartitionsCache {
+
+    private final Map<String, String[]> partitionsPerEventType = new ConcurrentHashMap<>();
```

Comment:
Then I would argue for us to implement an event type cache and move partitioning information to the event type. Wdyt?